Sources/ArcGISToolkit/Components/Offline/PreplannedMapModel.swift

```diff
@@ -86,9 +77,24 @@ class PreplannedMapModel: ObservableObject, Identifiable {
             observeJob(foundJob)
         } else if let mmpk = lookupMobileMapPackage() {
             Logger.offlineManager.debug("Found MMPK for area \(preplannedMapAreaID.rawValue, privacy: .public)")
+            status = .downloaded
+            Task.detached { await self.loadAndUpdateMobileMapPackage(mmpk: mmpk) }
+        }
+    }
+    
+    /// Tries to load a mobile map package and if successful, then updates state
+    /// associated with it.
+    private func loadAndUpdateMobileMapPackage(mmpk: MobileMapPackage) async {
+        do {
+            try await mmpk.load()
             mobileMapPackage = mmpk
             directorySize = FileManager.default.sizeOfDirectory(at: mmpkDirectoryURL)
```

Comment:
This was being set in multiple places. The loadAndUpdateMobileMapPackage consolidates it.